clippy_lints/src/redundant_async_block.rs

```diff
@@ -0,0 +1,51 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet};
+use rustc_ast::ast::*;
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `async` block that only returns `await` on a future.
+    ///
+    /// ### Why is this bad?
+    /// It is shorter to directly use the future.
+    ///
+    /// ### Example
+    /// ```rust
+    /// await {
+    ///     function_returning_a_future().await
```

Comment:
Might be good if we had an `async fn function_returning_a_future() {}` so the example actually compiles. That applies to both code snippets.

Reply:
Done